content/en/blog/2019/monitoring-external-service-traffic/index.md

```diff
@@ -0,0 +1,360 @@
+  In that case, the route configuration of every virtual route in Envoy is augmented to
+  add the `BlackHoleCluster` like this:
+
+  {{< text json >}}
+    {
+      "name": "block_all",
+      "domains": [
+        "*"
+      ],
+      "routes": [
+        {
+          "match": {
+            "prefix": "/"
+          },
+          "directResponse": {
+            "status": 502
+          }
+        }
+      ]
+    }
+  {{< /text >}}
+
+  In this case `502` response code is returned to the clients if no route exists
+  for a host.
+
+* **`PassthroughCluster`** - The `PassthroughCluster` is a fake virtual cluster created
+  in the Envoy configuration when `global.outboundTrafficPolicy.mode` is set to
+  `ALLOW_ANY`. In this mode, all traffic to any external service external is allowed.
+  To implement this, the default virtual outbound listener at `0.0.0.0:15001`
+  which uses `SO_ORIGINAL_DST`, is setup as a TCP Proxy with the `PassthroughCluster`
+  as the static cluster.
+  The configuration for the `PassthroughCluster` looks like this:
+
+  {{< text json >}}
+    {
+      "name": "PassthroughCluster",
+      "type": "ORIGINAL_DST",
+      "connectTimeout": "10s",
+      "lbPolicy": "ORIGINAL_DST_LB",
+      "circuitBreakers": {
+        "thresholds": [
+          {
+            "maxConnections": 102400,
+            "maxRetries": 1024
+          }
+        ]
+      }
+    }
+  {{< /text >}}
+
+  This cluster uses the `ORIGINAL_DST_LB` load balancing policy which configures
+  Envoy to send the traffic to the original destination i.e. passthrough.
+
+  Similar to the `BlackHoleCluster`, for every port/protocol based listener the
+  virtual route configuration is augmented to add the `PassthroughCluster` as the
+  default route:
+
+  {{< text json >}}
+    {
+      "name": "allow_any",
+      "domains": [
+        "*"
+      ],
+      "routes": [
+        {
+          "match": {
+            "prefix": "/"
+          },
+          "route": {
+            "cluster": "PassthroughCluster"
+          }
+        }
+      ]
+    }
+  {{< /text >}}
+
+Before release 1.3, there were no metrics reported or if metrics were reported
+there were no explicit labels set when traffic hit these clusters, resulting in
+lack of visibility in traffic flowing through the mesh.
+
+The next section covers how to take advantage of this enhancement as the metrics
+and labels emitted are conditional on whether the virtual outbound or explicit port/protocol
+listener is being hit.
+
+### Using the augmented metrics
+
+To capture all external service traffic in either of the cases (`BlackHole` or
+`Passthrough`), you will need to monitor `istio_requests_total` and
+`istio_tcp_connections_closed_total` metrics. Depending upon the Envoy listener
+type i.e. TCP proxy or HTTP proxy that gets invoked, one of these metrics
+will be incremented.
+
+Additionally, in case of a TCP proxy listener in order to see the IP address of
```

Comment:
do we want to add caveats about potential explosion in cardinality?  and it sounds like we probably want to create a doc on changing the metrics collection that we can link here... probably not a blocker for this blog post, though.

Reply:
about cardinality I can add a blurb.

I was thinking of creating another doc/blog for adding dimensions to Istio telemetry but like you said that's a follow-on and not a blocker for this blog.